### DISCUSSION TITLE
about ui components

### DISCUSSION THREAD

**Discussion by rraammiinn** [imba/imba#817] [Q&A]
why do it doesn't have any ui components libraries like vue .
it's biggest discouragement .
can be any way to translate components libraries of react or vue to imba components ?

**Comment by haikyuu** [comment]
Imba is quite compatible with web components. So it works quite well with web-component UI kits. Example: https://shoelace.style/

**Comment by familyfriendlymikey** [comment]
We have a work in progress at https://github.com/imba/imba/tree/master/packages/imba-ui

I think the reason we haven't made a comprehensive component library yet is because it's often just faster to make your own components, that's just the nature of Imba.

Which components are you looking for exactly?

**Reply by rraammiinn** [reply]
for someone like me who doesn't know css they're essential and some components like switch and more complex ones are very hard to create .
I like imba so much that I don't understand why it isn't more popular than typescript but because of components libraries I'm using vue .
I even can't make a dark theme for my existing app if the library doesn't provide it .
I can use css frameworks but they are limited to existing html elements .

**Reply by rraammiinn** [reply]
https://codepen.io

this site has very good examples for creating a great ui library .

**Comment by rraammiinn** [comment]
I only wana use css for layouts and positioning other than that I don't even change their colors .

**Comment by fatherofinvention** [comment]
@rraammiinn https://mint-lang.com which is similar to Imba has its own UI library: https://ui.mint-lang.com

**Comment by nirbhay12345** [comment]
Hi all, 
I think we can include this as a feature, also created an issue for similar topic: [Create Imba-UI Component docs](https://github.com/imba/imba/issues/899)

Let me know if we can proceed in this direction for imba-ui components
